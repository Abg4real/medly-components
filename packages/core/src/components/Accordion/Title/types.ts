import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp{
    color?: string;
    isActive?: boolean;
    onToggle?: () => void;
    bgColor?: string;

}


