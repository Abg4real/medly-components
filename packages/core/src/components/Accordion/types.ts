import { Props as TitleProps } from './Title';

export interface Props {
    Title?: React.SFC<TitleProps>;
    Content?: React.SFC;
}