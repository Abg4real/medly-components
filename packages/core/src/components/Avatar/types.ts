import { AvatarSizes } from '@medly-components/theme';
import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    size?: keyof AvatarSizes;
    textColor?: string;
    bgColor?: string;
}

export interface StyledProps extends Props {
    fontSize: string;
    avatarSize: string;
}
