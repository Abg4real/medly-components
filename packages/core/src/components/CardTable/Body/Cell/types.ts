import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    align: 'left' | 'center' | 'right';
    wrapText: false | true;
}
