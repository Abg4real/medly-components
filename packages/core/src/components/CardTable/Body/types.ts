import { ColumnConfig, Data } from '../types';

export interface Props {
    data: Data[];
    columns: ColumnConfig[];
    onRowClick?: (rowData: object) => void;
    rowClickDisableKey?: string;
    uniqueKeyName?: string;
    withWhiteBackground?: boolean;
}
