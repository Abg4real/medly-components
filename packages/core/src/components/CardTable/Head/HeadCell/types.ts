import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    withWhiteBackground: boolean;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    isActive: boolean;
    withWhiteBackground: boolean;
    flex: number;
    align: 'left' | 'center' | 'right';
}
