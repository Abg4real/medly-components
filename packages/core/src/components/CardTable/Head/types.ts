import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    withWhiteBackground: boolean;
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
