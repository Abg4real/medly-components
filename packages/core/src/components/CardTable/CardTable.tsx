import React, { SFC } from 'react';
import Body from './Body';
import * as Styled from './CardTable.styled';
import Head from './Head';
import { Props } from './types';

export const CardTable: SFC<Props> = React.memo(
    ({ withWhiteBackground, defaultSortField, defaultSortOrder, onSort, columns, ...restProps }) => (
        <Styled.Table>
            <Head {...{ defaultSortField, defaultSortOrder, onSort, columns, withWhiteBackground }} />
            <Body {...{ columns, withWhiteBackground, ...restProps }} />
        </Styled.Table>
    )
);

CardTable.displayName = 'CardTable';
CardTable.defaultProps = {
    withWhiteBackground: false
};
