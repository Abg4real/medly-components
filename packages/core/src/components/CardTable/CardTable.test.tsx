import { render } from '@test-utils';
import React from 'react';
import { CardTable } from './CardTable';
import { ColumnConfig } from './types';

describe('CardTable', () => {
    it('should render CardTable properly', () => {
        const onSort = jest.fn(),
            onRowClick = jest.fn(),
            columns: ColumnConfig[] = [
                { title: 'Patient Info', field: 'patientInfo', sort: true },
                { title: 'RxInfo', field: 'rxInfo', sort: true }
            ];
        const data = [
            {
                id: 1,
                patientInfo: 'Oli Bob',
                rxInfo: 'Metaformin',
                price: '535.76',
                status: 'Pending'
            }
        ];

        const { container } = render(
            <CardTable
                data={data}
                onSort={onSort}
                columns={columns}
                onRowClick={onRowClick}
                rowClickDisableKey="disable"
                uniqueKeyName="123"
            />
        );
        expect(container).toMatchSnapshot();
    });
});
