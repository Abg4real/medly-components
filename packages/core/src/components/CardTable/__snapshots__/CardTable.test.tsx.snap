// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable should render properly with grey background 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.8rem;
  margin-bottom: 0.8rem;
  background-color: white;
  border: 0.1rem solid #ffffff;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
}

.c6:hover {
  box-shadow: 0 0.4rem 0.8rem rgba(176,188,200,0.6);
}

.c6:nth-child(odd) {
  background: #ffffff;
}

.c6:nth-child(even) {
  background: #ffffff;
}

.c0 {
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: transparent;
}

.c5 {
  font-size: 1.6rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c5 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c2:hover .c3 {
  color: #1A6BF0;
}

.c2:hover::after {
  background-color: #1A6BF0;
}

.c2:hover .sc-AykKI * {
  fill: #1A6BF0;
}

.c2 .c3 {
  display: inline;
  cursor: default;
  color: #435465;
}

.c2 .sc-AykKI {
  cursor: pointer;
  padding-left: 1rem;
}

.c2 .sc-AykKI * {
  fill: #435465;
}

<div>
  <table
    class="c0"
  >
    <thead>
      <tr
        class="c1"
      >
        <th
          class="c2"
        >
          <h5
            class="c3 c4"
          >
            Patient Info
          </h5>
          <svg
            class="c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
        <th
          class="c2"
        >
          <h5
            class="c3 c4"
          >
            RxInfo
          </h5>
          <svg
            class="c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="c6"
      >
        <td
          class="c7"
        >
          <span
            class="c3 c8"
          >
            Oli Bob
          </span>
        </td>
        <td
          class="c7"
        >
          <span
            class="c3 c8"
          >
            Metaformin
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable should render properly with white background 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.8rem;
  margin-bottom: 0.8rem;
  background-color: white;
  border: 0.1rem solid #dfe4e9;
  box-shadow: none;
}

.c6:hover {
  box-shadow: 0px 0.2rem 0.4rem rgba(176,188,200,0.6);
}

.c6:nth-child(odd) {
  background: #F7F8F9;
}

.c6:nth-child(even) {
  background: #ffffff;
}

.c0 {
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: transparent;
}

.c5 {
  font-size: 1.6rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c5 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c2:hover .c3 {
  color: #1A6BF0;
}

.c2:hover::after {
  background-color: #1A6BF0;
}

.c2:hover .sc-AykKI * {
  fill: #1A6BF0;
}

.c2 .c3 {
  display: inline;
  cursor: default;
  color: #607890;
}

.c2 .sc-AykKI {
  cursor: pointer;
  padding-left: 1rem;
}

.c2 .sc-AykKI * {
  fill: #607890;
}

<div>
  <table
    class="c0"
  >
    <thead>
      <tr
        class="c1"
      >
        <th
          class="c2"
        >
          <h5
            class="c3 c4"
          >
            Patient Info
          </h5>
          <svg
            class="c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
        <th
          class="c2"
        >
          <h5
            class="c3 c4"
          >
            RxInfo
          </h5>
          <svg
            class="c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="c6"
      >
        <td
          class="c7"
        >
          <span
            class="c3 c8"
          >
            Oli Bob
          </span>
        </td>
        <td
          class="c7"
        >
          <span
            class="c3 c8"
          >
            Metaformin
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
